packages/forms-web-app/src/controllers/lpa-dashboard/appeal-details.js

```diff
@@ -14,50 +15,86 @@ const { getLPAUserFromSession } = require('../../services/lpa-user.service');
 const { isFeatureActive } = require('../../featureFlag');
 const { FLAG } = require('@pins/common/src/feature-flags');
 
+/**
+ * @typedef {import('../../lib/appeals-api-wrapper').documentMetaData} documentMetaData
+ */
+
+/**
+ * Retrieves specific data from a document's metadata.
+ * @param {documentMetaData | Array<documentMetaData> } docMetaData - Document MetaData or array of Document MetaData
+ * @returns {Object | Array<Object> } An object or array of objects containing extracted data from the document metadata.
+ * @property {string} filename - The name of the document file.
+ * @property {string} documentURI - The URI of the document.
+ */
+function getDocumentViewData(docMetaData) {
+	if (Array.isArray(docMetaData)) {
+		return docMetaData.map((document) => ({
+			filename: document.filename,
+			documentURI: document.documentURI
+		}));
+	}
+
+	return {
+		filename: docMetaData.filename,
+		documentURI: docMetaData.documentURI
+	};
+}
+
 const getAppealDetails = async (req, res) => {
 	const { id } = req.params;
 	const user = getLPAUserFromSession(req);
 	const caseReference = encodeURIComponent(id);
 	const appeal = await getAppealByLPACodeAndId(user.lpaCode, caseReference);
-	const applicationFormMetaData = await getAppealDocumentMetaData(
-		caseReference,
-		encodeURI('Planning application form')
-	);
-	const decisionLetterMetaData = await getAppealDocumentMetaData(
-		caseReference,
-		encodeURI('Decision notice')
-	);
-	const appealStatementMetaData = await getAppealDocumentMetaData(
-		caseReference,
-		encodeURI('Appeal Statement')
-	);
-	const supportingDocumentsMetaData = await getAppealDocumentMetaData(
-		caseReference,
-		encodeURI('Supporting Documents'),
-		true
-	);
-
-	const supportingDocuments = supportingDocumentsMetaData.map((document) => ({
-		filename: document.filename,
-		documentURI: document.documentURI
-	}));
 
 	const documents = {
-		applicationForm: {
-			filename: applicationFormMetaData.filename,
-			documentURI: applicationFormMetaData.documentURI
-		},
-		decisionLetter: {
-			filename: decisionLetterMetaData.filename,
-			documentURI: decisionLetterMetaData.documentURI
-		},
-		appealStatement: {
-			filename: appealStatementMetaData.filename,
-			documentURI: appealStatementMetaData.documentURI
-		},
-		supportingDocuments
+		applicationForm: null,
+		decisionLetter: null,
+		appealStatement: null,
+		supportingDocuments: null
 	};
 
+	try {
+		const applicationFormMetaData = await getAppealDocumentMetaData(
```

Comment:
My only suggestion would be, is there a way to make this use an inline function to wrap all the try catch?
Something like
const documents = {applicationForm: null,
    decisionLetter: null,
    appealStatement:null,
    supportingDocuments:null}

async function getMetadata(caseReference, uri){
    try {
            let docMetaData = await getAppealDocumentMetaData(
                caseReference,
                encodeURI(uri)
            );
            let view = getDocumentViewData(docMetaData);
            return view;
        } catch (err) {
        
            logger.error(err);
        }
    }

// then do something like this to set all the values one by one
documents.applicationForm = await getMetadata(caseReference, 'Planning application form')